### ISSUE TITLE
API-27: MP Reactive Messaging 1.0 Epic

### ISSUE THREAD

**Issue by tevans78** [OpenLiberty/open-liberty#5008]
Implement MicroProfile Reactive Messaging spec https://github.com/eclipse/microprofile-reactive-messaging.

The AHA feature for this is here: https://bigblue.aha.io/features/API-27

We will implement the MicroProfile Reactive Messaging specification as well as a
Kafka 'connector'. This connector must work equally as well with a Kafka queue
that is hosted as a native Confluent/Apache Kafka system or an IBM Event Streams
system whether on ICP or IBM cloud - depending on what is behind the configured
URL for the queues.

This feature depends on the feature MicroProfile reactive streams #4089

**Comment by marikaj123** [comment]
@tevans78 - Was the feature code beta enabled and included into the Liberty 19007 beta build?

**Comment by tevans78** [comment]
Yes, it was delivered as beta.

**Comment by tevans78** [comment]
@gscottj This feature doesn't have any UI. Please can you give accessibility approval?

**Comment by tevans78** [comment]
@jhanders34 This feature should not have any performance impact on anything else. Please can you give us performance approval? At some point in the future we should do some Kafka throughput testing; there may well be improvements we can make.

**Comment by gscottj** [comment]
@tevans78  Accessibility approval granted.  This feature does not have any user interface, therefore no accessibility verification testing is required.

**Comment by tevans78** [comment]
> @tevans78 - Does the feature require translation?

@marikaj123 Short answer: no
Longer answer: It only has a couple of NLS messages. They may have missed the last translation date (I'd have to double check) but it is really only a couple so no big deal.
The documentation for this release is simply a blog post so that doesn't need any translation either.

**Comment by tevans78** [comment]
@marikaj123 Correct, they were only for additional validation of some config options. They don't affect functionality at all.

**Comment by tevans78** [comment]
@donbourne 
Serviceability Approval Comment - Please answer the following questions for serviceability approval:

    WAD -- does the WAD identify the most likely problems customers will see and identify how the feature will enable them to diagnose and solve those problems without resorting to raising a PMR? Have these issues been addressed in the implementation?
_Yes - most issues that customers will face will actually come from their configuration of the Kafka Client. As the Kafka Client is not something we are shipping, this is no different to as if they were using the client directly and isn't really within our control. That said, we are including as much validation as we can along with debugging tips in the blog post. There will also be additional content later on in guides._

    Test and Demo -- As part of the serviceability process we're asking feature teams to test and analyze common problem paths for serviceability and demo those problem paths to someone not involved in the development of the feature (eg. L2, test team, or another development team).
    a) What problem paths were tested and demonstrated? _Sending messages to Kafka; Recieving messages from Kafka; and using connectors to manipulate and modify the messages in between._
    b) Who did you demo to? _Our local team and SVT_
    c) Do the people you demo'd to agree that the serviceability of the demonstrated problem scenarios is sufficient to avoid PMRs for any problems customers are likely to encounter, or that L2 should be able to quickly address those problems without need to engage L3? _SVT had a concern about the high level of logging which was coming out of the Kafka client. However, we determined that this was mainly down to an unusual usage pattern and could be switched off if required._

    SVT -- SVT team is often the first team to try new features and often encounters problems setting up and using them. Note that we're not expecting SVT to do full serviceability testing -- just to sign-off on the serviceability of the problem paths they encountered.
    a) Who conducted SVT tests for this feature? _Brian Hanczaryk_
    b) Do they agree that the serviceability of the problems they encountered is sufficient to avoid PMRs, or that L2 should be able to quickly address those problems without need to engage L3? _yes_

    Which L2 / L3 queues will handle PMRs for this feature? Ensure they are present in the contact reference file and in the queue contact summary, and that the respective L2/L3 teams know they are supporting it. Ask Don Bourne if you need links or more info. _CDI L3 queue_

**Comment by skasund** [comment]
Providing STE approval early. Please provide the slides as soon as possible. Thanks!

**Comment by ayoho** [comment]
FAT approved. The SOE had some hiccups getting some of the test containers to work correctly, but based on discussions between Dave and Andy it looks like the coverage is still good. Thanks everyone!

**Comment by chirp1** [comment]
Development is doing a blog post for 19009: OpenLiberty/open-liberty#6682 . Approving.

**Comment by lauracowen** [comment]
From @chirp1:
Here is the issue for a 19.0.0.9 blog post: OpenLiberty/open-liberty#6682 and the gist for the blog post https://gist.github.com/Azquelt/056e096f3879ed66f2adfc54a0b3b230 The feature is mpReactiveMessaging-1.0
Laura also opened #505 to collate all MP Config properties

**Comment by marikaj123** [comment]
Tagging with release:19009 since the code made it into the Gm build.

**Comment by marikaj123** [comment]
@tevans78 - Is the Epic ready to close?
